internal/utils/web.go

```diff
@@ -100,32 +100,51 @@ func AuthHandler(password string) http.Handler {
 			return // only POST is allowed
 		}
 
+		writeErr := func(msg string, statusCode int) {
+			w.Header().Set("Content-Type", "text/plain")
+			w.WriteHeader(statusCode)
+			w.Write([]byte(msg))
+		}
+
 		// parse validity
 		validityMS := req.FormValue("validity")
 		if validityMS == "" {
-			w.Header().Set("Content-Type", "text/plain")
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte("'validity' parameter is missing"))
+			writeErr("'validity' parameter is missing", http.StatusBadRequest)
 			return
 		}
 		var validity time.Duration
 		if _, err := fmt.Sscan(validityMS, &validity); err != nil {
-			w.Header().Set("Content-Type", "text/plain")
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte("failed to parse validity"))
+			writeErr("failed to parse validity", http.StatusBadRequest)
 			return
 		}
 		validity *= time.Millisecond
 
 		// generate token
 		token := authTokens.GenerateNew(validity)
 
+		// get host domain from header
+		var host string
```

Comment:
Perhaps we could use `X-Forwarded-Host` and instruct people to configure the reverse proxy in a way that it's being set to the original value in the `Host` header?

```go
host := req.Header.Get("X-Forwarded-Host")
if host == "" {
    host = req.Host // fallback
}
```